dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocityWebUtil.java

```diff
@@ -0,0 +1,151 @@
+                        }
+
+
+
+                    }
+                }
+            } catch (Exception e) {
+                Logger.debug(VelocityWebUtil.class, e.getMessage(), e);
+            }
+        }
+
+    }
+
+    public String mergeTemplate(String templatePath) throws ResourceNotFoundException, ParseErrorException, Exception {
+
+        VelocityEngine ve = VelocityUtil.getEngine();
+        Template template = null;
+        StringWriter sw = new StringWriter();
+        String threadName = Thread.currentThread()
+            .getName();
+        if (this.debug) {
+            Logger.info(VelocityWebUtil.class, _logVariable + ": " + templatePath);
+        }
+        try {
+            String newThreadName = (threadName.contains("{")) ? threadName.replaceAll("\\{[^\\}]*\\}", "{" + templatePath + "}")
+                    : threadName + " {" + templatePath + "}";
+            Thread.currentThread()
+                .setName(newThreadName);
+            template = ve.getTemplate(templatePath);
+
+            template.merge(ctx, sw);
+            return sw.toString();
+        } catch (ParseErrorException e) {
+            if (null != ctx) {
+
+                /*
+                 * In Edit mode we are allow to fail and be noisy, but on Preview and Live mode we
+                 * just want to continue with the render of the page.
+                 */
+                PageMode mode = (PageMode) ctx.get(WebKeys.PAGE_MODE_SESSION);
+
+                if (null == mode || mode != PageMode.EDIT_MODE) {
+                    Logger.error(this.getClass(), "Error parsing elements", e);
+                    return sw.toString();
+                }
+            }
+
+            throw e;
+        } finally {
+            Thread.currentThread()
+                .setName(threadName);
+        }
+
+
+    }
+
+
+    public void mergeTemplate(String templatePath, HttpServletResponse response)
+            throws ResourceNotFoundException, ParseErrorException, Exception {
+        VelocityEngine ve = VelocityUtil.getEngine();
+        Template template = null;
+        String threadName = Thread.currentThread()
+            .getName();
+        try {
+            Thread.currentThread()
+                .setName(threadName + " >" + templatePath);
+            if (this.debug) {
+                Logger.info(VelocityWebUtil.class, _logVariable + ": " + templatePath);
+            }
+
+            template = ve.getTemplate(templatePath);
+            template.merge(ctx, response.getWriter());
+        } finally {
+            Thread.currentThread()
+                .setName(threadName);
+        }
+    }
+
+
+
+    public boolean doesVelocityResourceExist(String templatePath) {
+        VelocityEngine ve = VelocityUtil.getEngine();
+        try {
+            ve.getTemplate(templatePath);
+        } catch (ResourceNotFoundException e) {
+            return false;
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)